### PR TITLE
Vcs minor clean ups

### PR DESCRIPTION
**Pull request by fviernau**
Minor non-functional changes in order to make it a bit easier working with the code as preparation for the JGit implementation.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/879)

### CODE REVIEW

File: downloader/src/main/kotlin/VersionControlSystem.kt [hunk 1]
```diff
@@ -220,7 +220,7 @@ abstract class VersionControlSystem {
     /**
      * Return the Java class name as a simple way to refer to the [VersionControlSystem].
      */
-    override fun toString(): String = javaClass.simpleName
+    fun getType(): String = javaClass.simpleName
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/VersionControlSystem.kt]: This should be a property: `val type = javaClass.simpleName`

File: downloader/src/main/kotlin/VersionControlSystem.kt [hunk 2]
```diff
@@ -230,7 +230,7 @@ abstract class VersionControlSystem {
         /**
          * Return a simple string representation for the VCS this working tree belongs to.
          */
-        fun getType() = this@VersionControlSystem.toString()
+        fun getType() = this@VersionControlSystem.getType()
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/VersionControlSystem.kt]: Unrelated to this change, but this should also be a property: `val type = this@VersionControlSystem.type`

File: downloader/src/main/kotlin/VersionControlSystem.kt (hunk 1 above)
Comment (sschuberth) [reply downloader/src/main/kotlin/VersionControlSystem.kt]: Also, this commit misses at least one call site of `toString()`:

https://github.com/heremaps/oss-review-toolkit/blob/b596eda3a757307b3a30c8aaad43d7aed0d76538/downloader/src/main/kotlin/Downloader.kt#L96

Please use the IntelliJ's "Find usages" (Alt-F7) to check. I'm not sure whether that also find implicit calls to `toString()` inside of string messages, though.

Comment (sschuberth) [reply downloader/src/main/kotlin/VersionControlSystem.kt]: > This should be a property: val type = javaClass.simpleName

I basically agree, but I'd prefer to do that separately. We have a bunch of more getters in here that could / should be properties, but for this commit I'd prefer to keep the diff as small as possible and not change semantics where not required.

File: downloader/src/main/kotlin/WorkingTree.kt [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+/**
+ * A class representing a local VCS working tree. The passed [workingDir] does not necessarily need to be the
+ * root directory of the tree. The root directory can be determined by calling [getRootPath].
+ */
+abstract class WorkingTree(val workingDir: File, val vcsType: String) {
+
+    /**
+     * Return a simple string representation for the VCS this working tree belongs to.
+     */
+    fun getType(): String = vcsType
```
[the remark this answers is not in the document]
Comment (fviernau) [reply downloader/src/main/kotlin/WorkingTree.kt]: Agree, but that's out of scope of this change right? (I'd fix that on top)

[the remark this answers is not in the document]
Comment (sschuberth) [reply downloader/src/main/kotlin/WorkingTree.kt]: Well, in this case it's an obvious simplification that became possible by your refactoring, so it's slight different from changing the other getter into properties, and personally I would have done this in this commit.

[the remark this answers is not in the document]
Comment (fviernau) [reply downloader/src/main/kotlin/WorkingTree.kt]: sorry, I fully agree with you. When writing my above comment I accidently assumed it's in context of a different commit.

**Pull request merged**